ui/widgets/calendar.js

```diff
@@ -230,146 +244,162 @@ return $.widget( "ui.calendar", {
+			title = $( "<div>", { role: "header", id: this.gridId + "-title" } ),
+			notice = $( "<span>" ).text( ", " + this._getTranslation( "datePickerRole" ) );
+
+		this._addClass( header, "ui-calendar-header", "ui-widget-header ui-helper-clearfix" )
+			._addClass( notice, null, "ui-helper-hidden-accessible" );
+
+		return header.append(
+			title
+				.append( this._buildTitle() )
+				.append( notice )
+		);
 	},
 
 	_buildTitle: function() {
-		return "<span class='ui-calendar-month'>" +
-			this.viewDate.monthName() +
-		"</span> " +
-		"<span class='ui-calendar-year'>" +
-			this.viewDate.year() +
-		"</span>";
+		var title = $( "<div>", { role: "alert", id: this.gridId + "-month-label" } ),
+			month = $( "<span>" ).text( this.viewDate.monthName() ),
+			year = $( "<span>" ).text( this.viewDate.year() );
+
+		this._addClass( title, "ui-calendar-title" )
+			._addClass( month, "ui-calendar-month" )
+			._addClass( year, "ui-calendar-year" );
+
+		return title
+			.append( month )
+			.append( " " )
+			.append( year );
 	},
 
 	_buildGrid: function() {
-		return "<table class='ui-calendar-calendar' role='grid' aria-readonly='true' " +
-			"aria-labelledby='" + this.gridId + "-month-label' tabindex='0' " +
-			"aria-activedescendant='" + this._getDayId( this.date ) + "'>" +
-			this._buildGridHeading() +
-			this._buildGridBody() +
-		"</table>";
+		var table = $( "<table>", {
+			role: "grid",
+			tabindex: 0,
+			"aria-readonly": true,
+			"aria-labelledby": this.gridId + "-month-label",
+			"aria-activedescendant": this._getDayId( this.date )
+		} );
+
+		this._addClass( table, "ui-calendar-calendar" );
+
+		return table
+			.append( this._buildGridHeading() )
+			.append( this._buildGridBody() );
 	},
 
 	_buildGridHeading: function() {
-		var cells = "",
+		var head = $( "<thead role='presentation'>" ),
+			week = $( "<th>" ),
+			row = $( "<tr role='row'>" ),
 			i = 0,
 			weekDayLength = this.viewDate.weekdays().length,
 			weekdays = this.viewDate.weekdays();
 
 		if ( this.options.showWeek ) {
-			cells += "<th class='ui-calendar-week-col'>" + this._getTranslation( "weekHeader" ) + "</th>";
+			this._addClass( week, "ui-calendar-week-col" );
+			row.append( week.text( this._getTranslation( "weekHeader" ) ) );
 		}
+
 		for ( ; i < weekDayLength; i++ ) {
-			cells += this._buildGridHeaderCell( weekdays[ i ] );
+			row.append( this._buildGridHeaderCell( weekdays[ i ] ) );
 		}
 
-		return "<thead role='presentation'>" +
-			"<tr role='row'>" + cells + "</tr>" +
-		"</thead>";
+		return head.append( row );
 	},
 
 	_buildGridHeaderCell: function( day ) {
-		return "<th role='columnheader' abbr='" + day.fullname + "' aria-label='" + day.fullname + "'>" +
-			"<span title='" + day.fullname + "'>" +
-				day.shortname +
-			"</span>" +
-		"</th>";
+		return $( "<th role='columnheader' abbr='" + day.fullname + "' aria-label='" + day.fullname + "'>" +
+			"<span title='" + day.fullname + "'>" + day.shortname + "</span>" +
+		"</th>" );
```

Comment:
I've converted all string concatenations where it was needed in order to use the classes option. I tried to convert as little as possible due to performance considerations. 

For example: day cell rendering is called quite a lot so this still uses string concatenation and has no support for `classes` option yet.

Do we want to use `$(elem, props)` for _all_ elements?